sdk/python/v1beta1/kubeflow/katib/api/katib_client_test.py

```diff
@@ -270,6 +291,217 @@ def create_experiment(
+        nproc_per_node=None,
+        pytorch_replica_specs={},
+        run_policy=None,
+    ):
+        self.elastic_policy = elastic_policy
+        self.nproc_per_node = nproc_per_node
+        self.pytorch_replica_specs = pytorch_replica_specs
+        self.run_policy = run_policy
+
+
+class KubeflowOrgV1PyTorchJob:
+    def __init__(
+        self,
+        api_version=None,
+        kind=None,
+        metadata=None,
+        spec=KubeflowOrgV1PyTorchJobSpec,
+        status=None,
+    ):
+        self.api_version = api_version
+        self.kind = kind
+        self.metadata = metadata
+        self.spec = spec
+        self.status = status
+
+
+test_tune_data = [
+    (
+        "not specify name",
+        {
+            "name": None,
+            "objective": lambda x: x,
+            "parameters": {"param": "value"},
+        },
+        ValueError,
+    ),
+    (
+        "set both options",
+        {
+            "name": "tune_test",
+            "objective": lambda x: x,
+            "model_provider_parameters": HuggingFaceModelParams(),
+        },
+        ValueError,
+    ),
+    (
+        "set no option",
+        {
+            "name": "tune_test",
+        },
+        ValueError,
+    ),
+    (
+        "set one option but missing parameters - only objective",
+        {
+            "name": "tune_test",
+            "objective": lambda x: x,
+        },
+        ValueError,
+    ),
+    (
+        "set one option but missing parameters - only parameters",
+        {
+            "name": "tune_test",
+            "parameters": {"param": "value"},
+        },
+        ValueError,
+    ),
+    (
+        "set one option but missing parameters - only model_provider_parameters",
+        {
+            "name": "tune_test",
+            "model_provider_parameters": HuggingFaceModelParams(),
+        },
+        ValueError,
+    ),
+    (
+        "set one option but missing parameters - only dataset_provider_parameters",
+        {
+            "name": "tune_test",
+            "dataset_provider_parameters": HuggingFaceDatasetParams(),
+        },
+        ValueError,
+    ),
+    (
+        "set one option but missing parameters - only trainer_parameters",
+        {
+            "name": "tune_test",
+            "trainer_parameters": HuggingFaceTrainerParams(),
+        },
+        ValueError,
+    ),
+    (
+        "invalid env_per_trial",
+        {
+            "name": "tune_test",
+            "objective": lambda x: x,
+            "parameters": {"param": "value"},
+            "env_per_trial": "invalid",
+        },
+        ValueError,
+    ),
+    (
+        "invalid model_provider_parameters",
+        {
+            "name": "tune_test",
+            "model_provider_parameters": "invalid",
+            "dataset_provider_parameters": HuggingFaceDatasetParams(),
+            "trainer_parameters": HuggingFaceTrainerParams(),
+        },
+        ValueError,
+    ),
+    (
+        "invalid dataset_provider_parameters",
```

Comment:
Do you also need test case for invalid trainer_parameters ?

Reply:
Previously, we didn’t include a type check for `trainer_parameters` in the `tune` API. You’re right that it’s important to do so. I’ve now added the type check for `trainer_parameters` in the tune API, along with the corresponding unit test.
https://github.com/kubeflow/katib/blob/b25f7ba37a60710191666b936ed47598c2de4ac2/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L603-L606
https://github.com/kubeflow/katib/blob/b25f7ba37a60710191666b936ed47598c2de4ac2/sdk/python/v1beta1/kubeflow/katib/api/katib_client_test.py#L473-L481